### ISSUE TITLE
Bottom pixel row of weapon sprite cut off

### ISSUE THREAD

**Issue by Blzut3** [ECWolfEngine/ECWolf#206]
**[Original report](https://bitbucket.org/ecwolf/ecwolf/issues/206) by Anonymous.**

----------------------------------------

Running the game with the statusbar always seems to incorrectly cut off the bottom row of pixels of the weapon sprite.

If you play in "full" view with no statusbar then it is sometimes visible (yet partially cut off).
It displays at 640*480 but even at other resolutions it is partially cut off (1024*768, 1280*960 etc). Even then it is partially cut off in height compared to the other "pixel rows" of the sprite.

(This happens with all sprites).

(Aside - For me, for aspect ratio correction, i would suggest rendering square pixels and doubling out to an integer ratio with no filtering and THEN after that, changing the pixel height to 1.2 X the height with linear filtering to correct the pixel aspect but still keep things sharp.
From what i can see of how it looks now, it might be that you are just using no filtering / point filtering for the whole thing?).

Hope that helps?
Thanks for your time :)

**Comment by Blzut3** [comment]
Issue #147 was marked as a duplicate of this issue.

**Comment by Blzut3** [comment]
This is a duplicate of #147, but since you provided screenshots I'll use this one instead.

As for the aspect correction note, what you're asking for is super sampling which can't really be done effectively in paletted 8bpp.

**Comment by Blzut3** [comment]
**Original comment by Anonymous.**

----------------------------------------

Sorry for the extremely late reply.

I understand what you are saying about super sampling - i had envisaged doing it the opposite way - rendering at the highest amount that would "fit" into the fullscreen res (pre correction) and then linear scaling upwards to aspect correct.

That said, you're right in that it wouldn't work in paletted 8bpp.

Could you render out in 8bpp pre-aspect correction and then take that output and aspect correct it in standard 24 (or 32) bit with linear filtering?
So basically as a kind of post processing?

**Comment by Blzut3** [comment]
That would be possible with some work.  Essentially render the 2D elements to a separate surface at a different resolution and then blend them together at the end in high/true color.  A *little* bit of the ground work is laid out already since the HUD is actually drawn with an API from ZDoom that can be hardware accelerated.  However, honestly there are other things I would rather spend my time on.

**Comment by Blzut3** [comment]
**Original comment by Anonymous.**

----------------------------------------

Fair enough, thanks for the reply!

**Comment by Blzut3** [comment]
Issue #233 was marked as a duplicate of this issue.
